include/openPMD/backend/Attribute.hpp

```diff
@@ -180,27 +185,42 @@ auto doConvert(T *pv) -> U
             }
             return res;
         }
-
-        throw std::runtime_error(
-            "getCast: no vector to array conversion possible.");
+        else
+        {
+            throw std::runtime_error(
+                "getCast: no vector to array conversion possible.");
+        }
     }
     // conversion cast: turn a single value into a 1-element vector
     else if constexpr (auxiliary::IsVector_v<U>)
     {
         if constexpr (std::is_convertible_v<T, typename U::value_type>)
         {
-            U res;
+            U res{};
             res.reserve(1);
             res.push_back(static_cast<typename U::value_type>(*pv));
             return res;
         }
-
-        throw std::runtime_error(
-            "getCast: no scalar to vector conversion possible.");
+        else
+        {
+            throw std::runtime_error(
+                "getCast: no scalar to vector conversion possible.");
+        }
     }
-
-    (void)pv;
-    throw std::runtime_error("getCast: no cast possible.");
+    else
+    {
+        throw std::runtime_error("getCast: no cast possible.");
+    }
+#if defined(__INTEL_COMPILER)
+/*
+ * ICPC has trouble with if constexpr, thinking that return statements are
+ * missing afterwards. Deactivate the warning.
+ * Note that putting a statement here will not help to fix this since it will
+ * then complain about unreachable code.
+ * https://community.intel.com/t5/Intel-C-Compiler/quot-if-constexpr-quot-and-quot-missing-return-statement-quot-in/td-p/1154551
+ */
+#pragma warning(disable : 1011)
```

Comment:
Since this is a `.hpp` file, do you need to `pop` this again?

Reply:
Yes, I need to reset it to default again after this

Reply:
done in latest commit :)